standalone/pruntime/enclave/src/contracts/mod.rs

```diff
@@ -50,6 +63,190 @@ where
+    Con: NativeContract<Cmd = Cmd, Event = Event, QReq = QReq, QResp = QResp>,
+{
+    contract: Con,
+    send_mq: MessageChannel,
+    cmd_rcv_mq: PeelingReceiver<PushCommand<Cmd>, CmdWrp, CmdPlr>,
+    event_rcv_mq: PeelingReceiver<Event, EventWrp, EventPlr>,
+    ecdh_key: KeyPair,
+}
+
+impl<Con, Cmd, CmdWrp, CmdPlr, Event, EventWrp, EventPlr, QReq, QResp>
+    NativeCompatContract<Con, Cmd, CmdWrp, CmdPlr, Event, EventWrp, EventPlr, QReq, QResp>
+where
+    Cmd: Decode + BindTopic + Debug + Send + Sync,
+    CmdWrp: Decode + BindTopic + Debug + Send + Sync,
+    CmdPlr: Peeler<Wrp = CmdWrp, Msg = PushCommand<Cmd>>,
+    Event: Decode + BindTopic + Debug + Send + Sync,
+    EventWrp: Decode + BindTopic + Debug + Send + Sync,
+    EventPlr: Peeler<Wrp = EventWrp, Msg = Event>,
+    QReq: Serialize + DeserializeOwned + Debug,
+    QResp: Serialize + DeserializeOwned + Debug,
+    Con: NativeContract<Cmd = Cmd, Event = Event, QReq = QReq, QResp = QResp> + Send + Sync,
+    PushCommand<Cmd>: Decode + BindTopic + Debug,
+{
+    pub fn new(
+        contract: Con,
+        send_mq: MessageChannel,
+        cmd_rcv_mq: PeelingReceiver<PushCommand<Cmd>, CmdWrp, CmdPlr>,
+        event_rcv_mq: PeelingReceiver<Event, EventWrp, EventPlr>,
+        ecdh_key: KeyPair,
+    ) -> Self {
+        NativeCompatContract {
+            contract,
+            send_mq,
+            cmd_rcv_mq,
+            event_rcv_mq,
+            ecdh_key,
+        }
+    }
+}
+
+impl<Con, Cmd, CmdWrp, CmdPlr, Event, EventWrp, EventPlr, QReq, QResp> Contract
+    for NativeCompatContract<Con, Cmd, CmdWrp, CmdPlr, Event, EventWrp, EventPlr, QReq, QResp>
+where
+    Cmd: Decode + BindTopic + Debug + Send + Sync,
+    CmdWrp: Decode + BindTopic + Debug + Send + Sync,
+    CmdPlr: Peeler<Wrp = CmdWrp, Msg = PushCommand<Cmd>> + Send + Sync,
+    Event: Decode + BindTopic + Debug + Send + Sync,
+    EventWrp: Decode + BindTopic + Debug + Send + Sync,
+    EventPlr: Peeler<Wrp = EventWrp, Msg = Event> + Send + Sync,
+    QReq: Serialize + DeserializeOwned + Debug,
+    QResp: Serialize + DeserializeOwned + Debug,
+    Con: NativeContract<Cmd = Cmd, Event = Event, QReq = QReq, QResp = QResp> + Send + Sync,
+    PushCommand<Cmd>: Decode + BindTopic + Debug,
+{
+    fn id(&self) -> ContractId {
+        self.contract.id()
+    }
+
+    fn handle_query(
+        &self,
+        origin: Option<&runtime::AccountId>,
+        req: OpaqueQuery,
+    ) -> Result<OpaqueReply, OpaqueError> {
+        serde_json::to_value(
+            self.contract.handle_query(
+                origin,
+                deopaque_query(req)
+                    .map_err(|_| error_msg("Malformed request"))?
+                    .request,
+            ),
+        )
+        .map_err(|_| error_msg("serde_json serilize failed"))
+    }
+
+    fn process_events(&mut self, env: &mut ExecuteEnv) {
+        let storage = env.storage;
+        let keystore = |topic: &phala_mq::Path| {
```

Comment:
Not a perfect name. `keystore` is usually the store for private key. It's better to call it `key_registry` or `key_map`.